src/concatenate.jl

```diff
@@ -125,16 +159,56 @@ Base.copy(concat::Concatenated) = copyto!(similar(concat), concat)
 # default falls back to replacing interface with Nothing
 # this permits specializing on typeof(dest) without ambiguities
 # Note: this needs to be defined for AbstractArray specifically to avoid ambiguities with Base.
-@inline Base.copyto!(dest::AbstractArray, concat::Concatenated) =
-  copyto!(dest, convert(Concatenated{Nothing}, concat))
+@inline function Base.copyto!(dest::AbstractArray, concat::Concatenated)
+  return copyto!(dest, convert(Concatenated{Nothing}, concat))
+end
+
+_copy_or_fill!(A, inds, x) = fill!(view(A, inds...), x)
```

Comment:
I'm definitely okay with copying all of that here, but probably we could put a comment to refer to that and explain why? Also gives us a proper reference if it ever changes.

Reply:
Good point, I'll add that.